common/pinot/pinotQueryValidator.go

```diff
@@ -334,11 +334,20 @@ func (qv *VisibilityQueryValidator) processCustomKey(expr sqlparser.Expr) (strin
 	if !ok {
 		return "", errors.New("invalid comparison expression, right")
 	}
-	colValStr := string(colVal.Val)
 
 	// get the value type
 	indexValType := common.ConvertIndexedValueTypeToInternalType(valType, log.NewNoop())
 
+	// if it is dataTime, then check if it is time.Time() type
+	if indexValType == types.IndexedValueTypeDatetime {
```

Comment:
can we move this to L358, so if it is type IndexedValueTypeDatetime, we trim then return processCustomNum

Reply:
There'll be an extra step to convert the result to a string then. `colValStr := string(colVal.Val)`